### ISSUE TITLE
inplace ifft(fft(volume)) produces inconsistent results

### ISSUE THREAD

**Issue by psteinb** [StephanPreibisch/FourierConvolutionCUDALib#13]
I stumbled upon a numerical deficiency that cufft appears to have (which I already reported to nvidia):

```
$ ./tests/test_how_cufft_works --log_level=message
inplace    shape(x,y,z)=19, 17, 13      l2norm = 4510.09 #yikes!
inplace    shape(x,y,z)=16, 16, 16      l2norm = 5.33185e-07
inplace    shape(x,y,z)=27, 27, 27      l2norm = 4.26473e-06
inplace    shape(x,y,z)=25, 25, 25      l2norm = 8.05152e-06
inplace    shape(x,y,z)=14, 14, 14      l2norm = 3.12369e-06
inplace    shape(x,y,z)=128, 128, 128   l2norm = 5.01017e-05
outofplace shape(x,y,z)=19, 17, 13      l2norm = 0.00314017 #yikes!
outofplace shape(x,y,z)=16, 16, 16      l2norm = 5.33185e-07
outofplace shape(x,y,z)=27, 27, 27      l2norm = 4.26473e-06
outofplace shape(x,y,z)=25, 25, 25      l2norm = 8.05152e-06
outofplace shape(x,y,z)=14, 14, 14      l2norm = 3.12369e-06
outofplace shape(x,y,z)=128, 128, 128   l2norm = 5.01017e-05
```

For the above, I basically created a 3D volume with arbitrary input and performed ifft(fft(volume)). I then compared the result with the original. fftw does not show this behavior! I'll post any news from nvidia to this issue before closing it. I still hope that this is due to a bug in my code.

**Comment by psteinb** [comment]
@maweigert shared some code with me that I'll introduce to this package in a dedicated branch ticket13_cufft_inplace_garbage.

**Comment by psteinb** [comment]
phew, it was me all along. so what was going on? 
I stumbled over the following in the [cufft docs](http://docs.nvidia.com/cuda/cufft/#multi-dimensional):
`cufftComplex odata[N1][N2][N3/2+1];`
when the creating of complex buffers for real-2-complex transforms is discussed. 

_Disclaimer: Lecturing Alert_
In C style row major order, if an array is allocated the native way
`cufftComplex data[4][3][2];`
you get 4 consecutive planes, each made of 3 lines made of 2 "pixels" in memory! 

So I played around and looked at the [fttw docs](http://www.fftw.org/fftw3_doc/Multi_002dDimensional-DFTs-of-Real-Data.html#Multi_002dDimensional-DFTs-of-Real-Data). There it's crystal clear, only individual lines in memory are to be padded when doing inplace transforms (and out-of-place accordingly as above). However, the cufft documentation refers to plan creation like this:

```
cufftResult 
    cufftPlan3d(cufftHandle *plan, int nx, int ny, int nz, cufftType type);
...
nx  The transform size in the x dimension 
```

So I was confused and believed they meant N3 from above. Nope, they meant N1. :( I'll update my bug report to nvidia accordingly. Added the fix to this library and all tests given L2 norms as expected.

On top, my examples included "unsymmetric shaped" stacks only for "19, 17, 13", the other volumes were all symmetric "16, 16, 16", this is why I focused on this without thinking of the implementation. @maweigert motivated me to revisit that assumption and he was right.
